android/src/main/java/org/mozilla/mozstumbler/client/mapview/maplocation/MultiSourceLocationListener.java

```diff
@@ -0,0 +1,94 @@
+        @Override
+        public void receivedLocation() {
+            // Once the gps begins receiving locations, stop the network listener
+            UserPositionUpdateManager.enableLocationListener(mLocationManager, false, mNetworkLocationListener);
+        }
+    };
+
+    private final MapUpdatingLocationListener mNetworkLocationListener;
+    private final MapUpdatingLocationListener mGpsLocationListener;
+    private final WeakReference<MapFragment> mMapFragment;
+
+    private final GpsStatus.Listener mSatelliteListener = new GpsStatus.Listener() {
+        public void onGpsStatusChanged(int event) {
+            if (event == GpsStatus.GPS_EVENT_SATELLITE_STATUS && mLocationManager != null) {
+                GpsStatus gpsStatus = mLocationManager.getGpsStatus(null);
+                Iterable<GpsSatellite> sats = gpsStatus.getSatellites();
+                int satellites = 0;
+                int fixes = 0;
+                for (GpsSatellite sat : sats) {
+                    satellites++;
+                    if (sat.usedInFix()) {
+                        fixes++;
+                    }
+                }
+                if (mMapFragment.get() != null) {
+                    mMapFragment.get().updateGPSInfo(satellites, fixes);
+                }
+
+                if (fixes < GPSScanner.MIN_SAT_USED_IN_FIX) {
+                    notEnoughSatellites();
+                }
+            }
+        }
+    };
+
+    MultiSourceLocationListener(LocationManager manager, MapFragment mapFragment) {
+        mLocationManager = manager;
+        mMapFragment = new WeakReference<MapFragment>(mapFragment);
+        mNetworkLocationListener = new NetworkMapUpdatingLocationListener(mapFragment, null);
+        mGpsLocationListener = new GPSMapUpdatingLocationListener(mapFragment, mReceivedGpsLocation);
+    }
+
+    public void start() {
+        mLocationManager.addGpsStatusListener(mSatelliteListener);
+        setLocationUpdatesEnabled(true);
+    }
+
+    public void stop() {
+        mLocationManager.removeGpsStatusListener(mSatelliteListener);
+        setLocationUpdatesEnabled(false);
+    }
+
+    private  void setLocationUpdatesEnabled(boolean enable) {
+        if (mLocationManager == null) {
+            return;
+        }
+
+        if (enable) {
+            // enable NetworkLocationListener first because GpsLocationListener will disable it
+            UserPositionUpdateManager.enableLocationListener(mLocationManager, true, mNetworkLocationListener);
+            UserPositionUpdateManager.enableLocationListener(mLocationManager, true, mGpsLocationListener);
+        } else {
+            // disable GpsLocationListener first because it might enable NetworkLocationListener
+            UserPositionUpdateManager.enableLocationListener(mLocationManager, false, mGpsLocationListener);
+            UserPositionUpdateManager.enableLocationListener(mLocationManager, false, mNetworkLocationListener);
+        }
```

Comment:
You could now remove this method and merge it into `start` and `stop`, because it's only called from these methods, once with parameter `true`, and from the other with parameter `false`.